store/src/lib.rs

```diff
@@ -236,4 +283,21 @@ impl StoreConfig {
             StoreConfig::Directory { path } => directory::initialize(path),
         }
     }
+
+    pub fn initialize_explicit_mode<OT, K, V, MKT>(
+        &self,
+        mode: DirectoryStorageMode,
+    ) -> Result<Box<dyn Store<OT, K, V, MKT>>, StoreError>
+    where
+        OT: StoreOpenMode + 'static,
+        // K and V are supersets of the possible requirements for the different implementations
+        K: Eq + std::hash::Hash + Send + Sync + std::string::ToString + std::str::FromStr + 'static,
+        V: Send + Sync + Clone + Serializable + 'static,
+        MKT: crate::MetadataLocalKey + 'static,
+    {
+        match self {
+            #[cfg(feature = "directory")]
+            StoreConfig::Directory { path } => directory::initialize_explicit_mode(path, mode),
+        }
+    }
```

Comment:
this kind of seem lot of duplicate code - I feel everything around xattrs support should really just be hidden within the directory code w/o touching anything else